From patchwork Mon Aug 16 03:14:37 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Samuel Holland <samuel@sholland.org>
X-Patchwork-Id: 1517002
X-Patchwork-Delegate: trini@ti.com
Return-Path: <u-boot-bounces@lists.denx.de>
X-Original-To: incoming@patchwork.ozlabs.org
Delivered-To: patchwork-incoming@bilbo.ozlabs.org
Authentication-Results: ozlabs.org;
 spf=pass (sender SPF authorized) smtp.mailfrom=lists.denx.de
 (client-ip=85.214.62.61; helo=phobos.denx.de;
 envelope-from=u-boot-bounces@lists.denx.de; receiver=<UNKNOWN>)
Authentication-Results: ozlabs.org;
	dkim=pass (2048-bit key;
 unprotected) header.d=sholland.org header.i=@sholland.org header.a=rsa-sha256
 header.s=fm3 header.b=t3ljhGk8;
	dkim=pass (2048-bit key;
 unprotected) header.d=messagingengine.com header.i=@messagingengine.com
 header.a=rsa-sha256 header.s=fm3 header.b=uNvUJ1OR;
	dkim-atps=neutral
Received: from phobos.denx.de (phobos.denx.de [85.214.62.61])
	(using TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits)
	 key-exchange X25519 server-signature RSA-PSS (4096 bits))
	(No client certificate requested)
	by ozlabs.org (Postfix) with ESMTPS id 4Gnznx3VLCz9sWw
	for <incoming@patchwork.ozlabs.org>; Mon, 16 Aug 2021 13:16:05 +1000 (AEST)
Received: from h2850616.stratoserver.net (localhost [IPv6:::1])
	by phobos.denx.de (Postfix) with ESMTP id A723982F03;
	Mon, 16 Aug 2021 05:14:55 +0200 (CEST)
Authentication-Results: phobos.denx.de;
 dmarc=none (p=none dis=none) header.from=sholland.org
Authentication-Results: phobos.denx.de;
 spf=pass smtp.mailfrom=u-boot-bounces@lists.denx.de
Authentication-Results: phobos.denx.de;
	dkim=pass (2048-bit key;
 unprotected) header.d=sholland.org header.i=@sholland.org
 header.b="t3ljhGk8";
	dkim=pass (2048-bit key;
 unprotected) header.d=messagingengine.com header.i=@messagingengine.com
 header.b="uNvUJ1OR";
	dkim-atps=neutral
Received: by phobos.denx.de (Postfix, from userid 109)
 id 266A982F2D; Mon, 16 Aug 2021 05:14:49 +0200 (CEST)
X-Spam-Checker-Version: SpamAssassin 3.4.2 (2018-09-13) on phobos.denx.de
X-Spam-Level: 
X-Spam-Status: No, score=-2.0 required=5.0 tests=BAYES_00,DKIM_SIGNED,
 DKIM_VALID,DKIM_VALID_AU,RCVD_IN_MSPIKE_H3,RCVD_IN_MSPIKE_WL,
 SPF_HELO_PASS autolearn=ham autolearn_force=no version=3.4.2
Received: from out2-smtp.messagingengine.com (out2-smtp.messagingengine.com
 [66.111.4.26])
 (using TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits))
 (No client certificate requested)
 by phobos.denx.de (Postfix) with ESMTPS id 63D1582EF4
 for <u-boot@lists.denx.de>; Mon, 16 Aug 2021 05:14:43 +0200 (CEST)
Authentication-Results: phobos.denx.de;
 dmarc=none (p=none dis=none) header.from=sholland.org
Authentication-Results: phobos.denx.de;
 spf=pass smtp.mailfrom=samuel@sholland.org
Received: from compute6.internal (compute6.nyi.internal [10.202.2.46])
 by mailout.nyi.internal (Postfix) with ESMTP id B635F5C00E3;
 Sun, 15 Aug 2021 23:14:39 -0400 (EDT)
Received: from mailfrontend1 ([10.202.2.162])
 by compute6.internal (MEProxy); Sun, 15 Aug 2021 23:14:39 -0400
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=sholland.org; h=
 from:to:cc:subject:date:message-id:mime-version
 :content-transfer-encoding; s=fm3; bh=3cL3ljkfWRKJ0oTpqHHqZYS/mY
 wGTQkcL6Fb+B3z+2I=; b=t3ljhGk8B7EcNeI6IcLB0rfZ3vKdD4lgmglup9+OHV
 yA0SUy4h20hdjClpozBIO+u5b1S3JUALMPa+YSlZC9bTRyJDZSuH5ZzaSo3sIXxc
 mwC8Gdqf1zQjWsrC5mDfzbhWHv8K7JBlIf92xAIdw79HZWz6QtZBUkOVXzRjSLSH
 YAI/xnJg7JeFOpfLY9D6+Aqssczd0EAof7cbpNjzG0NL++MA93EK181YUATXJGVY
 ZNcgGGSRPPv91/dG9+4R9R3/M8A0QLFJ3BFNthwaPp/EucBHPTd8fNg7Eq4i0N+l
 ZxHU+ltyu4XDzQ1CD9vrL7djTDwNSmsHEpVhy4YWOm+w==
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=
 messagingengine.com; h=cc:content-transfer-encoding:date:from
 :message-id:mime-version:subject:to:x-me-proxy:x-me-proxy
 :x-me-sender:x-me-sender:x-sasl-enc; s=fm3; bh=3cL3ljkfWRKJ0oTpq
 HHqZYS/mYwGTQkcL6Fb+B3z+2I=; b=uNvUJ1ORC+IsfJnlk+yHZja/8LjVMt4S4
 R5e1Q05YkYWNM8YHHponnnAZef9NUHyMPpsSRTu2NjOVrmyD1USWsb04zeXm/6Qr
 WFwy0lxnPkTUdBaXLo6tHxpl1vvPSZNxJkjiPhxvkJvWFtIYzCNl3OhTdR59ZIn9
 oL05LexKrPIBd8uBf+DycuIrXoJu9DNXNJBo3YN5fz+zV0vyqSKOwh85Be5yyugu
 nqs1GE4/tWTLkMh3vulsYKhPtXEzoKtlUJ/f/k9Drq56mADSmYsUKzGbE5hTCcD+
 2RsUAT5dR7LkeezEIQq/IgbiMOYIyKibJ84sCIMIm6+nAqXXZqMGw==
X-ME-Sender: <xms:H9gZYbzk0hNOlRgE2j_IvZrTRzHk8tWsUVgiaK9z60ul-jKalKh3LA>
 <xme:H9gZYTRokhdf989hIPSTLf3pDmqwPSED3VhAMuP5BSqa8kmk_S4magKCX12xtQJLk
 b04x5QGqk84cdrENA>
X-ME-Received: 
 <xmr:H9gZYVVuRbIucCOR1kENuas_DLhM-OkSy6enlm2VKP5DRDvMtvfA0kJtcEJKSM7-7rfOUzKW5oycZjX06s5zhTMu8oPiNQuccsVuCLVoNh5qhaWa7PAgWULM_kXbzQ9lvMZ9iQ>
X-ME-Proxy-Cause: 
 gggruggvucftvghtrhhoucdtuddrgedvtddrledtgdeilecutefuodetggdotefrodftvf
 curfhrohhfihhlvgemucfhrghsthforghilhdpqfgfvfdpuffrtefokffrpgfnqfghnecu
 uegrihhlohhuthemuceftddtnecusecvtfgvtghiphhivghnthhsucdlqddutddtmdenuc
 fjughrpefhvffufffkofgggfestdekredtredttdenucfhrhhomhepufgrmhhuvghlucfj
 ohhllhgrnhguuceoshgrmhhuvghlsehshhholhhlrghnugdrohhrgheqnecuggftrfgrth
 htvghrnhepieetkefhheduudfgledtudefjeejfeegveehkeeufffhhfejkeehiefftdev
 tdevnecuvehluhhsthgvrhfuihiivgeptdenucfrrghrrghmpehmrghilhhfrhhomhepsh
 grmhhuvghlsehshhholhhlrghnugdrohhrgh
X-ME-Proxy: <xmx:H9gZYVjE7_me_UWUZUmcf06LNu7qOC1Q8kbnM7g0UYspWEMoFzuZqw>
 <xmx:H9gZYdBt4WLlKJRb_3mM60RN5ydioNFzcAoarIVh5bMyuRCPzX_sCw>
 <xmx:H9gZYeI9S78feyXOmkBxLco5eElfKYKSAf1a0e4Mg96e0PeZAzdrlw>
 <xmx:H9gZYcO9dJkieQKxJZUEn2BRGy03t4g-PSn8yCOUhxYj5VbH9ABdpA>
Received: by mail.messagingengine.com (Postfix) with ESMTPA; Sun,
 15 Aug 2021 23:14:38 -0400 (EDT)
From: Samuel Holland <samuel@sholland.org>
To: Tom Rini <trini@konsulko.com>,
	u-boot@lists.denx.de
Cc: Jagan Teki <jagan@amarulasolutions.com>,
 Andre Przywara <andre.przywara@arm.com>,
 Samuel Holland <samuel@sholland.org>
Subject: [PATCH] ARM: Prevent the compiler from using NEON registers
Date: Sun, 15 Aug 2021 22:14:37 -0500
Message-Id: <20210816031437.15856-1-samuel@sholland.org>
X-Mailer: git-send-email 2.31.1
MIME-Version: 1.0
X-BeenThere: u-boot@lists.denx.de
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: U-Boot discussion <u-boot.lists.denx.de>
List-Unsubscribe: <https://lists.denx.de/options/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=unsubscribe>
List-Archive: <https://lists.denx.de/pipermail/u-boot/>
List-Post: <mailto:u-boot@lists.denx.de>
List-Help: <mailto:u-boot-request@lists.denx.de?subject=help>
List-Subscribe: <https://lists.denx.de/listinfo/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=subscribe>
Errors-To: u-boot-bounces@lists.denx.de
Sender: "U-Boot" <u-boot-bounces@lists.denx.de>
X-Virus-Scanned: clamav-milter 0.103.2 at phobos.denx.de
X-Virus-Status: Clean

For ARMv8-A, NEON is standard, so the compiler can use it even when no
special target flags are provided. For example, it can use stores from
NEON registers to zero-initialize large structures. GCC 11 decides to
do this inside the DRAM init code for the Allwinner H6, which breaks
boot on that platform, as NEON is not available in SPL. Fix this by
restricting the compiler to using GPRs only, not vector registers.

Signed-off-by: Samuel Holland <samuel@sholland.org>
---
 arch/arm/config.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/config.mk b/arch/arm/config.mk
index 16c63e12667..964c6b026ec 100644
--- a/arch/arm/config.mk
+++ b/arch/arm/config.mk
@@ -25,6 +25,7 @@ endif
 
 PLATFORM_RELFLAGS += -fno-common -ffixed-r9
 PLATFORM_RELFLAGS += $(call cc-option, -msoft-float) \
+		     $(call cc-option,-mgeneral-regs-only) \
       $(call cc-option,-mshort-load-bytes,$(call cc-option,-malignment-traps,))
 
 # LLVM support
